src/main/java/org/wso2/carbon/module/swiftiso20022/mt/parsers/MTParser.java

```diff
@@ -0,0 +1,247 @@
+
+        if (fields.isEmpty()) {
+
+            String errorMessage = String.format(
+                    ConnectorConstants.ERROR_BLOCK_INVALID_FORMAT, ConnectorConstants.USER_HEADER_BLOCK);
+            log.error(errorMessage);
+            throw new MTMessageParsingException(errorMessage);
+        }
+
+        UserHeaderBlock userHeaderBlock = new UserHeaderBlock();
+
+        for (String key : fields.keySet()) {
+
+            switch (key) {
+
+                case Field103.TAG:
+                    userHeaderBlock.setServiceIdentifier(Field103.parse(fields.get(key)));
+                    break;
+
+                case Field106.TAG:
+                    userHeaderBlock.setMessageInputReference(Field106.parse(fields.get(key)));
+                    break;
+
+                case Field108.TAG:
+                    userHeaderBlock.setMessageUserReference(Field108.parse(fields.get(key)));
+                    break;
+
+                case Field111.TAG:
+                    userHeaderBlock.setServiceTypeIdentifier(Field111.parse(fields.get(key)));
+                    break;
+
+                case Field113.TAG:
+                    userHeaderBlock.setBankingPriority(Field113.parse(fields.get(key)));
+                    break;
+
+                case Field115.TAG:
+                    userHeaderBlock.setAddresseeInformation(Field115.parse(fields.get(key)));
+                    break;
+
+                case Field119.TAG:
+                    userHeaderBlock.setValidationFlag(Field119.parse(fields.get(key)));
+                    break;
+
+                case Field121.TAG:
+                    userHeaderBlock.setEndToEndReference(Field121.parse(fields.get(key)));
+                    break;
+
+                case Field165.TAG:
+                    userHeaderBlock.setPaymentReleaseInformation(Field165.parse(fields.get(key)));
+                    break;
+
+                case Field423.TAG:
+                    userHeaderBlock.setBalanceCheckpoint(Field423.parse(fields.get(key)));
+                    break;
+
+                case Field424.TAG:
+                    userHeaderBlock.setRelatedReference(Field424.parse(fields.get(key)));
+                    break;
+
+                case Field433.TAG:
+                    userHeaderBlock.setSanctionsScreeningInformation(Field433.parse(fields.get(key)));
+                    break;
+
+                case Field434.TAG:
+                    userHeaderBlock.setPaymentControlsInformation(Field434.parse(fields.get(key)));
+                    break;
+
+                // If tag didn't match any of the above tags, it is not allowed
+                default:
+
+                    String errorMessage = String.format(ConnectorConstants.ERROR_FIELD_NOT_ALLOWED_IN_BLOCK,
+                            key, ConnectorConstants.USER_HEADER_BLOCK);
+                    log.error(errorMessage);
+                    throw new MTMessageParsingException(errorMessage);
```

Comment:
Shall we remove the extra lines inside the switch block

Reply:
Removed in a82da3f30afc3995c983ec8d68a04acfc363b6c0